### PR TITLE
Bundler, Rspec upgrade, and some other changes

### PR DESCRIPTION
**Pull request by stve**
This is primarily some simple changes.  I've added bundler, removed jeweler, and upgraded dependencies to hashie 1.0.0 and rspec 2.5.0

The larger change was to adjust the locations of some files to be a bit more in line with how the Mash class is defined (basically hashie/rash instead of all changes encapsulated in rash.rb).  I've persisted rash.rb to require 'hashie/rash' for backwards compatability.

I know your readme says not to fudge with the gemspec, but I needed to in order to ditch jeweler.

Thanks.

### CODE REVIEW

**Comment by tcocca** [comment]
Did you actually change anything here functionality wise?  What is the reason to remove Jeweler?  I can understand the namespacing and that is a good change to be more in line with Hashie::Mash but I don't get the removal of Jeweler.  Please elaborate.

Glad somebody else found this useful (I hate camelCasing!)

**Comment by stve** [comment]
I've used it plenty on a number of different projects.  Camel-case was driving me nuts too and I'm glad I stumbled across this  when I went looking for a solution some time back.

No, there were no functional changes, I did adjust the indentation in the the main class but didn't touch a line of actual code.  As for Jeweler, it isn't nearly as useful as it once was when github served gems and there was a need for it's utility.  Most of the value deriven from Jeweler has been replaced by Bundler's tasks now which I included in the Rakefile.

File: rash.gemspec [hunk 1]
```diff
@@ -1,58 +1,29 @@
-# Generated by jeweler
-# DO NOT EDIT THIS FILE DIRECTLY
-# Instead, edit Jeweler::Tasks in Rakefile, and run the gemspec command
 # -*- encoding: utf-8 -*-
+$:.push File.expand_path("../lib", __FILE__)
+require "rash/version"
 
 Gem::Specification.new do |s|
   s.name = %q{rash}
   s.version = "0.2.0"
```
Comment (joshk) [inline rash.gemspec]: This isn't needed anymore as Hashie::Rash::VERSION has been added a couple of lines down

**Comment by joshk** [comment]
+1 to all the changes made in this pull request, especially the removal of "require 'rubygems'"!!

**Comment by sferik** [comment]
:+1:

If these get merged and released, I can replace the custom Rash class in faraday_middleware, which gets used in many popular gems (e.g. the twitter gem).

**Comment by tcocca** [comment]
Ok, I am planning on merging this stuff hopefully this weekend.  I guess ill
just release a 0.3 version after that. Thanks guys.

~ tom

On Apr 2, 2011 1:02 PM, "sferik" <
reply@reply.github.com>
wrote:

:+1:

If these get merged and released, I can replace the custom Rash class in
faraday_middleware, which gets used in many popular gems (e.g. the twitter
gem).

##

**Comment by tcocca** [comment]
Here ya go guys, get it while its hot.

gem install rash

Thanks for the contributions, I didn't even know anybody else was using this gem, glad you guys find it useful.

~ Tom

**Comment by sferik** [comment]
Awesome! Thanks, Tom.

**Comment by sferik** [comment]
By the way, have you talked to Michael about getting this behavior merged into a future version of Hashie?

**Comment by tcocca** [comment]
A long time ago, see this issue:  https://github.com/intridea/hashie/issues/closed#issue/7

Michael and Mislav kinda shot it down, so I created a rash gem, it seemed silly at the time to have a full gem for such a simple thing but its been fine so far.  Maybe they'll link to it in the README for hashie.

What are you using rash on anyway?

**Comment by sferik** [comment]
See the thread on this commit: https://github.com/pengwynn/faraday_middleware/commit/3130404a16022b36f3c9786fc05e6c75bfc0a746#comments

**Comment by tcocca** [comment]
Cool, well let me know if there are any problems with the new stuff and I'll take a look.

**Pull request merged**